packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1904,6 +1910,15 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         // but would not modify contents details
         let message = { ...messageArg };
 
+        if (message.metadata?.compressed) {
+            const contents = IsoBuffer.from(message.contents.contents, "base64");
+            const decompressedMessage = decompress(contents);
+            const intoString = new TextDecoder().decode(decompressedMessage);
+            const asObj = JSON.parse(intoString);
+            message.contents.contents = asObj;
+            message.metadata.compressed = false;
```

Comment:
I'm not sure why this is necessary. From testing, tests were failing because we were processing the message again after it was decompressed and trying to decompress it again. Any insight into why the message would come through multiple times?

Reply:
Consider making it part of unpackRuntimeMessage() function below (if possible).
We have tools (like replay-tool) that poke at op stream, and they would be screwed with compression. But it uses unpackRuntimeMessage(), so it decompression is there, it would not break the tool

Reply:
Answering you question - tests (especially snapshot tests) will feed same op to multiple container instances. The right fix is somewhere at the top of op processing (in Container?) to deep-clone op before processing. But the cheap trick above with shallow copy (see line 1930) solves most of the problems and it's mostly sufficient (we parse content in Container conditionally only if it's a string). The issue is - metadata is an object, so that shallow copy does not help.

I think it's worth adding deep copy in Container. But only if strings are not duplicated (I believe they are immutable objects, so I hope deep clone logic only makes a copy of a structure of op). Alternatively, we can special case metadata - it's the only complex property on an object.

I think Curtis had some issues with lodash's deepClone(). I do not remember what it was - possibly big impact on bundle size?